base/src/main/java/com/smartdevicelink/managers/screen/SoftButtonObject.java

```diff
@@ -309,4 +309,20 @@ public void setOnEventListener(OnEventListener onEventListener) {
          */
         void onUpdate();
     }
+
+    /**
+     * @param o
+     * @return true if two SoftButtonObjects have the same name.
+     */
+    @Override
+    public boolean equals(Object o) {
+        if(o == null){
+            return false;
+        }
+        SoftButtonObject softButtonObject = (SoftButtonObject) o;
+        if (softButtonObject.getName().equals(this.getName())) {
```

Comment:
Potential NPE here. If the `name` instance is null it will throw an NPE. Need to check that it is not null before calling the `equals` method on it.

Reply:
I have added fix for this.